app/bundles/EmailBundle/EventListener/CampaignSubscriber.php

```diff
@@ -307,8 +307,31 @@ public function onCampaignTriggerActionSendEmailToContact(PendingEvent $event):
         }
 
         if ('marketing' == $type) {
+			
+            // Prepare a list of email IDs that we are going to check in contact's history, including the translations.
+            $emailIdsToCheck = [$emailId];
```

Comment:
You can use $email->getRelatedEntityIds() to find all related Ids https://github.com/mautic/mautic/blob/1db63347702e57b488a0e6432887d268df95a35f/app/bundles/CoreBundle/Entity/VariantEntityTrait.php#L250